node/network/dispute-distribution/src/sender/mod.rs

```diff
@@ -397,8 +398,10 @@ impl RateLimit {
 						target: LOG_TARGET,
 						?occasion,
 						?candidate_hash,
+						?num_wakes,
 						"Sending rate limit hit, slowing down requests"
 					);
+					num_wakes += 1;
```

Comment:
nit: Shouldn't this be incremented before the log? Or the first log will say 0 wakes which could be misleading.

Reply:
I would argue it is correct: The very first time we call into `poll` there was no `wake` yet. So 0 is what I would expect here.

I added this, to see whether really `limit` is called multiple times on the same candidate or if we get woken by the executer multiple times with result `Pending`. I already checked the logs and it is indeed the latter. (`num_wakes` is 0 and then 1 the next time.) Not sure where this redundant wake is coming from, but that is what is happening and it should be harmless (except for the wasted cycles).